tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/languageserver/LanguageServerConnectionProvider.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.ide.project.ui.languageserver;
+
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.project.core.modules.BladeCLIException;
+
+import java.io.File;
+
+import java.util.Arrays;
+import java.util.Properties;
+
+import org.eclipse.lsp4e.server.ProcessOverSocketStreamConnectionProvider;
+import org.eclipse.wst.server.core.util.SocketUtil;
```

Comment:
Lets depend on our own SocketUTil class. So we don't have to depend on WTP which may be not in Liferay IDE 4.0

Reply:
will do